### PR TITLE
[Fix] Fix get_valid_count flaky test for cuda

### PR DESCRIPTION
**Pull request by Laurawly**
Turned on get_valid_count test for cuda in topi.  Used atomic operations in this fix to replace previous block sync method.
Tested on V100 and T4 GPUs.

@trevor-m @kevinthesun @yzhliu Could you review?

### CODE REVIEW

File: topi/python/topi/cuda/nms.py [hunk 1]
```diff
@@ -28,27 +28,46 @@
...
+
+tvm.target.intrin.register_intrin_rule(
+    "cuda", "atomicAdd", cuda_atomicAdd_rule, override=True)
+
+
+def atomicAdd(x, y):
+    return tvm.call_pure_intrin(y.dtype, "atomicAdd", x, y)
+
+
+def get_valid_counts_ir(data, valid_count, Flag, score_threshold, id_index, score_index):
```
Comment (trevor-m) [inline topi/python/topi/cuda/nms.py]: Change variable/function names to fit conventions
`Flag` -> `flag`
`atomicAdd` -> `atomic_add`
etc

**Comment by Laurawly** [comment]
ping @kevinthesun @yzhliu

**Comment by yzhliu** [comment]
Thanks @Laurawly @trevor-m

**Comment by masahi** [comment]
@Laurawly I still get a flaky failure from get_valid_count test in my PR
https://ci.tvm.ai/blue/organizations/jenkins/tvm/detail/PR-4964/8/pipeline/246

**Comment by Laurawly** [comment]
@masahi That’s weird. You can comment off the test for now and I’ll try to reproduce it on my end.

**Comment by masahi** [comment]
@Laurawly Also reported at https://github.com/apache/incubator-tvm/pull/4931#issuecomment-594748097

**Comment by Laurawly** [comment]
> @Laurawly Also reported at a different PR [#4931 (comment)](https://github.com/apache/incubator-tvm/pull/4931#issuecomment-594306992)

Yeah, pls feel free to comment out the test: https://github.com/apache/incubator-tvm/blob/master/topi/tests/python/test_topi_vision.py#L106

**Comment by masahi** [comment]
fortunately my PR is green now without commenting out :) If the next open PR by somebody else sees the same problem, I'll ask him/her to comment it out.

**Pull request merged**